### PR TITLE
Remote debug

### PR DESCRIPTION
**Pull request by patricklee2**
fix to enable remote debugging with using npm start or custom start file

### CODE REVIEW

File: 10.1.0/startup/generateStartupCommand.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
```
Comment (patricklee2) [inline 10.1.0/startup/generateStartupCommand.js]: we can improve the readability and maintainability of this file by using some functions, but at a later date

File: 10.1.0/startup/node-wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+# check if inspect option is used
+# hope we didnt match too much, should improve on this
+args_contain_inspect=false
+for arg in "$@"; do
+  case "$arg" in
+    --inspect*)
+      args_contain_inspect=true
+      ;;
+    --debug*)
```
Comment (patricklee2) [inline 10.1.0/startup/node-wrapper.sh]: we should improve the matches to avoid wildcards

File: 10.1.0/Dockerfile [hunk 3]
```diff
@@ -23,6 +23,7 @@ RUN npm install -g pm2 \
      && apk add tcptraceroute \
      && apk add bash \
      && cd /opt/startup \
+     && npm install \
```
Comment (patricklee2) [inline 10.1.0/Dockerfile]: ensure dependencies for default site are installed

**Comment by shaunakv1** [comment]
@patricklee2 sounds very useful. Can you please provide a brief explanation of how to use this in actual app? say a vanilla express app. Thanks!

**Comment by patricklee2** [comment]
Our goal is to eventually support
`https://medium.com/@auchenberg/introducing-remote-debugging-of-node-js-apps-on-azure-app-service-from-vs-code-in-public-preview-9b8d83a6e1f0`, but without the need for  adding ` "start_azure_debug": "node --inspect=0.0.0.0:$APPSVC_TUNNEL_PORT index.js"` to package.json.  

Once the deployments are complete we should be able to use remote debugging with custom startup scripts.

**Pull request merged**